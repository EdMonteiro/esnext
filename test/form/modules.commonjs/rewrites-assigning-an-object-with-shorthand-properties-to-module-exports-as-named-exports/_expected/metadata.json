{
  "modules.commonjs": {
    "imports": [],
    "exports": [
      {
        "type": "named-export",
        "bindings": [
          {
            "localName": "a",
            "exportName": "a"
          },
          {
            "localName": "b",
            "exportName": "b"
          }
        ],
        "node": {
          "type": "ExpressionStatement",
          "expression": {
            "type": "AssignmentExpression",
            "operator": "=",
            "left": {
              "type": "MemberExpression",
              "object": {
                "type": "Identifier",
                "name": "module",
                "typeAnnotation": null
              },
              "property": {
                "type": "Identifier",
                "name": "exports",
                "typeAnnotation": null
              },
              "computed": false
            },
            "right": {
              "type": "ObjectExpression",
              "properties": [
                {
                  "type": "ObjectProperty",
                  "computed": false,
                  "key": {
                    "type": "Identifier",
                    "name": "a",
                    "typeAnnotation": null
                  },
                  "value": {
                    "type": "Identifier",
                    "name": "a",
                    "typeAnnotation": null
                  },
                  "shorthand": true,
                  "decorators": null
                },
                {
                  "type": "ObjectProperty",
                  "computed": false,
                  "key": {
                    "type": "Identifier",
                    "name": "b",
                    "typeAnnotation": null
                  },
                  "value": {
                    "type": "Identifier",
                    "name": "b",
                    "typeAnnotation": null
                  },
                  "shorthand": true,
                  "decorators": null
                }
              ]
            }
          }
        }
      }
    ],
    "directives": []
  }
}