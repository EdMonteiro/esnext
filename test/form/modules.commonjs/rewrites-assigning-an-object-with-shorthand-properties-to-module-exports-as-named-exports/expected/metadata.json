{
  "modules.commonjs": {
    "imports": [],
    "exports": [
      {
        "type": "named-export",
        "bindings": [
          {
            "exportName": "a",
            "localName": "a"
          },
          {
            "exportName": "b",
            "localName": "b"
          }
        ],
        "node": {
          "type": "ExpressionStatement",
          "expression": {
            "type": "AssignmentExpression",
            "operator": "=",
            "left": {
              "type": "MemberExpression",
              "computed": false,
              "object": {
                "type": "Identifier",
                "name": "module"
              },
              "property": {
                "type": "Identifier",
                "name": "exports"
              }
            },
            "right": {
              "type": "ObjectExpression",
              "properties": [
                {
                  "type": "Property",
                  "computed": false,
                  "kind": "init",
                  "method": false,
                  "shorthand": true,
                  "key": {
                    "type": "Identifier",
                    "name": "a"
                  },
                  "value": {
                    "type": "Identifier",
                    "name": "a"
                  }
                },
                {
                  "type": "Property",
                  "computed": false,
                  "kind": "init",
                  "method": false,
                  "shorthand": true,
                  "key": {
                    "type": "Identifier",
                    "name": "b"
                  },
                  "value": {
                    "type": "Identifier",
                    "name": "b"
                  }
                }
              ]
            }
          }
        }
      }
    ],
    "directives": []
  }
}